doc/man7/openssl-quic.pod

```diff
@@ -0,0 +1,681 @@
+
+Applications which wish to implement QUIC-specific protocols should be aware of
+the APIs listed under B<QUIC-SPECIFIC APIS> which provide access to
+QUIC-specific functionality. For example, L<SSL_stream_conclude(3)> can be used
+to indicate the end of the sending part of a stream, and L<SSL_shutdown_ex(3)>
+can be used to provide a QUIC application error code when closing a connection.
+
+=head1 QUIC-SPECIFIC APIS
+
+This section details new APIs which are directly or indirectly related to QUIC.
+For details on the operation of each API, see the referenced man pages.
+
+The following SSL APIs are new but relevant to both QUIC and DTLS:
+
+=over 4
+
+=item L<SSL_get_tick_timeout(3)>
+
+Determines when the QUIC implementation should next be woken up via a call to
+L<SSL_tick(3)> (or another I/O function such as L<SSL_read(3)> or
+L<SSL_write(3)>), if ever.
+
+This can also be used with DTLS and supersedes L<DTLSv1_get_timeout(3)> for new
+usage.
+
+=item L<SSL_tick(3)>
+
+This is a non-specific I/O operation which makes a best effort attempt to
+perform any pending I/O or timeout processing. It can be used to advance the
+QUIC state machine by processing incoming network traffic, generating outgoing
+network traffic and handling any expired timeout events. Most other I/O
+functions on an SSL object, such as L<SSL_read(3)> and L<SSL_write(3)>
+implicitly perform ticking of the SSL object, so calling this function is only
+needed if no other I/O function is to be called.
+
+This can also be used with DTLS and supersedes L<DTLSv1_handle_timeout(3)> for
+new usage.
+
+=back
+
+The following SSL APIs are specific to QUIC:
+
+=over 4
+
+=item L<SSL_set_blocking_mode(3)>, L<SSL_get_blocking_mode(3)>
+
+Configures whether blocking semantics are used at the application level. This
+determines whether calls to functions such as L<SSL_read(3)> and L<SSL_write(3)>
+will block.
+
+=item L<SSL_get_rpoll_descriptor(3)>, L<SSL_get_wpoll_descriptor(3)>
+
+These functions facilitate operation in nonblocking mode.
+
+When an SSL object is being used with an underlying network read BIO which
+supports polling, L<SSL_get_rpoll_descriptor(3)> outputs an OS resource which
+can be used to synchronise on network readability events which should result in
+a call to L<SSL_tick(3)>. L<SSL_get_wpoll_descriptor(3)> works in an analogous
+fashion for the underlying network write BIO.
+
+The poll descriptors provided by these functions need only be used when
+L<SSL_net_read_desired(3)> and L<SSL_net_write_desired(3)> return 1, respectively.
+
+=item L<SSL_net_read_desired(3)>, L<SSL_net_write_desired(3)>
+
+These functions facilitate operation in nonblocking mode and are used in
+conjunction with L<SSL_get_rpoll_descriptor(3)> and
+L<SSL_get_wpoll_descriptor(3)> respectively. They determine whether the
+respective poll descriptor is currently relevant for the purposes of polling.
+
+=item L<SSL_set_initial_peer_addr(3)>, L<SSL_get_initial_peer_addr(3)>
```

Comment:
`SSL_get_initial_peer_addr` does not exist but `SSL_set_initial_peer_addr` does. Do we plan to add it?

Reply:
We might want to add an accessor for this, but I've updated this to match the reality for now.